### PR TITLE
Puzzle CRUD: Rework unordered puzzle logic

### PR DESCRIPTION
**Pull request by amoliksingh**
Loom video: https://www.loom.com/share/3a842811226745e2885fde9cafc3ce06

Basically, this PR reworks the unordered puzzle to use add and delete, and also ensures that when deleting images, they are in fact also deleted from S3

### CODE REVIEW

**Comment by ahmedhamodi** [comment]
Is this good for me to review @amoliksingh?

**Comment by amoliksingh** [comment]
> Is this good for me to review @amoliksingh?

Yes

File: src/Components/Admin/ObjectEditor/unorderedpuzzle.js [hunk 1]
```diff
@@ -158,18 +149,35 @@ export default function UnorderedPuzzle(props) {
...
+                    <IconButton
+                        className={props.classes.addButton}
+                        aria-label="add"
+                        onClick={() => {
+                            addImage();
+                        }}
+                    >
+                        <AddIcon />
+                    </IconButton>
+                </div>
```
Comment (ahmedhamodi) [inline src/Components/Admin/ObjectEditor/unorderedpuzzle.js:180]: So you know how there is a disabled prop on the delete button when # images == 2, should we have a disabled prop on add button when # images == 5?

**Review by ahmedhamodi — COMMENTED** [review]
Code looks good, only 1 question. Will test out locally now

File: src/Components/Admin/ObjectEditor/unorderedpuzzle.js (hunk 1 above)
Comment (amoliksingh) [reply src/Components/Admin/ObjectEditor/unorderedpuzzle.js:180]: Good idea, I think we probably should

Comment (ahmedhamodi) [reply src/Components/Admin/ObjectEditor/unorderedpuzzle.js:180]: Can you add that and I'll re-review

Comment (amoliksingh) [reply src/Components/Admin/ObjectEditor/unorderedpuzzle.js]: Added

**Review by ahmedhamodi — APPROVED** [review]
LGTM

**Pull request merged**